kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/PromiseFactory.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PromiseFactory {
+
+    private final ScheduledExecutorService executorService;
+    private final long timeout;
+    private final TimeUnit timeoutUnit;
+    private final Logger logger;
+
+    public PromiseFactory(ScheduledExecutorService executorService, long timeout, TimeUnit timeoutUnit, String loggerName) {
```

Comment:
* Why passing the logger name, rather than the logger itself?
* Is it the case that we want the `timeout` to be scoped to the factory, rather than per-promise? In more sophisticated uses you might want to define a timeout that applies over a sequence of timeoutable actions,  so you need to compute the remaining timeout and supply that for each subsequent action.

Reply:
I passed the name rather than the logger as I can't think of another example where I've seen loggers passed around so it felt less surprising, happy to be persuaded otherwise.

Current usage calls for a factory wide logger so that's all I did for now, I also wasn't convinced by the api of passing the logger at the calls. Again happy to be persuaded otherwise. 

I'm not sure we have use cases yet for request processing limits or global timeouts. Id rather refactor this when we do than try and design for a problem we don't have now.